tracer/src/Datadog.Trace/Util/SpanIdGenerator.cs

```diff
@@ -64,11 +64,16 @@ public static SpanIdGenerator ThreadInstance
 
         public ulong CreateNew()
         {
-            long high = _random.Next(int.MinValue, int.MaxValue);
-            long low = _random.Next(int.MinValue, int.MaxValue);
+            long value;
+            do
+            {
+                long high = _random.Next(int.MinValue, int.MaxValue);
+                long low = _random.Next(int.MinValue, int.MaxValue);
 
-            // Concatenate both values, and truncate the 32 top bits from low
-            var value = high << 32 | (low & 0xFFFFFFFF);
+                // Concatenate both values, and truncate the 32 top bits from low
+                value = high << 32 | (low & 0xFFFFFFFF);
+            }
+            while (value == 0);
 
             return (ulong)value & 0x7FFFFFFFFFFFFFFF;
```

Comment:
This doesn't _actually_ prevent `CreateNew()` returning zero due to the bitwise-AND we do in the return statement; if `value == 0x8000000000000000` then this still returns 0 😉 

Better to move the AND inside the loop I think (note you need to make `value` a `ulong` for this to compile:

```suggestion
                // Concatenate both values, and truncate the 32 top bits from low
                // The MSB is set to 0, as traceIDs are 63 bit (for interoperability reasons)
                value = (ulong)(high << 32 | (low & 0xFFFFFFFF)) & 0x7FFFFFFFFFFFFFFF;
            }
            while (value == 0);

            return value;
```

Reply:
true!, thank you. Done in [ae3563b](https://github.com/DataDog/dd-trace-dotnet/pull/3033/commits/ae3563b9bd4c965399aad15322e0fbafd125527f)